src/python/pants/task/task.py

```diff
@@ -651,6 +653,20 @@ def determine_target_roots(self, goal_name):
     # of e.g. `./pants --changed-parent=HEAD list` (w/ no changes) returning an empty result.
     return self.context.target_roots
 
+  def tgt_closure_has_native_sources(self):
```

Comment:
This should probably not go on `Task`: the only use of `self` is to call `self.context.targets`, but if a caller were to call that method directly, they could then call your utility function with the list of targets to check.

Reply:
(to be clear, all `self.context.targets(predicate)` is doing is doing `filter(predicate, all_targets)`)